### PR TITLE
docs: translate Referencing Values with Refs page

### PR DESCRIPTION
**Pull request by irsyaadbp**
Closes https://github.com/reactjs/id.react.dev/issues/387

### CODE REVIEW

**Review by r17x — CHANGES_REQUESTED** [review]
Untuk menjaga konsistensi, gunakan `*` dari pada `_` untuk tulisan yang *italic*

File: src/content/learn/referencing-values-with-refs.md [hunk 1]
```diff
@@ -1,50 +1,49 @@
 ---
-title: 'Referencing Values with Refs'
+title: 'Mereferensikan Nilai dengan Refs'
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:2]: `dengan` vs `menggunakan`, thoughts?

File: src/content/learn/referencing-values-with-refs.md [hunk 2]
```diff
@@ -1,50 +1,49 @@
...
-- How to use refs safely
+- Bagaimana cara menambahkan *ref* pada komponen Anda
+- Bagaimana cara memperbarui nilai *ref*
+- Perbedaan antara *refs* dan *state*
+- Bagaimana cara menggunakan *refs* dengan aman
 
 </YouWillLearn>
 
-## Adding a ref to your component {/*adding-a-ref-to-your-component*/}
+## Menambahkan *ref* pada komponen Anda {/*adding-a-ref-to-your-component*/}
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:20]:
```suggestion
## Menambahkan sebuah *ref* pada komponen Anda {/*adding-a-ref-to-your-component*/}
```

File: src/content/learn/referencing-values-with-refs.md [hunk 3]
```diff
@@ -1,50 +1,49 @@
...
 
-Inside your component, call the `useRef` Hook and pass the initial value that you want to reference as the only argument. For example, here is a ref to the value `0`:
+Di dalam komponen Anda, panggil `useRef` *Hook* dan berikan nilai awal yang Anda ingin referensikan sebagai satu-satunya argumen. Misalnya, berikut adalah *ref* yang mempunyai nilai `0`:
 
 ```js
 const ref = useRef(0);
 ```
 
-`useRef` returns an object like this:
+`useRef` mengembalikan sebuah objek seperti:
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:34]:
```suggestion
`useRef` mengembalikan sebuah objek seperti ini:
```

File: src/content/learn/referencing-values-with-refs.md [hunk 4]
```diff
@@ -55,33 +54,33 @@ export default function Counter() {
 
   function handleClick() {
     ref.current = ref.current + 1;
-    alert('You clicked ' + ref.current + ' times!');
+    alert('Anda mengeklik ' + ref.current + ' kali!');
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:57]:
```suggestion
    alert('Anda mengklik ' + ref.current + ' kali!');
```

File: src/content/learn/referencing-values-with-refs.md [hunk 5]
```diff
@@ -55,33 +54,33 @@ export default function Counter() {
...
 }
 ```
 
 </Sandpack>
 
-The ref points to a number, but, like [state](/learn/state-a-components-memory), you could point to anything: a string, an object, or even a function. Unlike state, ref is a plain JavaScript object with the `current` property that you can read and modify.
+*Ref* tersebut mengacu pada sebuah angka, tetapi, seperti [*state*](/learn/state-a-components-memory), Anda juga dapat mengacu pada tipe data apapun: sebuah *string*, objek, atau bahkan sebuah fungsi. Berbeda dengan *state*, *ref* adalah sebuah objek JavaScript biasa dengan properti `current` yang dapat Anda baca dan ubah nilainya.
 
-Note that **the component doesn't re-render with every increment.** Like state, refs are retained by React between re-renders. However, setting state re-renders a component. Changing a ref does not!
+Perhatikan bahwa **komponen tidak rerender setiap kali nilai pada *ref* ditambahkan.** Seperti *state*, nilai dari *refs* akan tetap disimpan atau dipertahankan oleh React saat rerender terjadi. Namun mengubah *state* akan memicu rerender pada komponen, sementara *ref* tidak akan melakukannya!
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:72]:
```suggestion
Perhatikan bahwa **komponen tidak di-*render* ulang setiap kali nilai pada *ref* ditambahkan.** Seperti *state*, nilai dari *refs* akan tetap disimpan atau dipertahankan oleh React saat *render* ulang terjadi. Namun mengubah *state* akan memicu *render* ulang pada komponen, sementara *ref* tidak akan melakukannya!
```

**Review by arfianadam — COMMENTED** [review]
1st review.

File: src/content/learn/referencing-values-with-refs.md (hunk 4 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:57]: saya kemarin nyoba nyari di kbbi nemunya `mengeklik` jadi akhirnya itu yang dipakai
https://kbbi.kemdikbud.go.id/entri/mengeklik

File: src/content/learn/referencing-values-with-refs.md (hunk 5 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:72]: Noted, *rerender* change to *render* ulang

File: src/content/learn/referencing-values-with-refs.md (hunk 1 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:2]: Iya kemarin bingung mau `dengan` atau `menggunakan`, tapi sepertinya memang lebih nyambung kalo`menggunakan`

File: src/content/learn/referencing-values-with-refs.md [hunk 6]
```diff
@@ -55,33 +54,33 @@ export default function Counter() {
...
 
-## Example: building a stopwatch {/*example-building-a-stopwatch*/}
+## Contoh: Membangun *stopwatch* {/*example-building-a-stopwatch*/}
 
-You can combine refs and state in a single component. For example, let's make a stopwatch that the user can start or stop by pressing a button. In order to display how much time has passed since the user pressed "Start", you will need to keep track of when the Start button was pressed and what the current time is. **This information is used for rendering, so you'll keep it in state:**
+Anda dapat menggabungkan *refs* dan *state* dalam satu komponen. Sebagai contoh, mari buat *stopwatch* yang dapat dijalankan atau dihentikan oleh pengguna dengan menekan tombol. Untuk menampilkan berapa waktu yang telah berlalu sejak pengguna menekan tombol "Mulai", Anda perlu melacak kapan tombol "Mulai" ditekan dan waktu saat ini. **Informasi ini digunakan untuk *rendering*, sehingga Anda akan menyimpannya dalam *state*:**
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:76]:
```suggestion
Anda dapat menggabungkan *refs* dan *state* dalam satu komponen. Sebagai contoh, mari buat *stopwatch* yang dapat dijalankan atau dihentikan oleh pengguna dengan menekan sebuah tombol. Untuk menampilkan berapa waktu yang telah berlalu sejak pengguna menekan tombol "Mulai", Anda perlu melacak kapan tombol "Mulai" ditekan dan waktu saat ini. **Informasi ini digunakan untuk *rendering*, sehingga Anda akan menyimpannya dalam *state*:**
```

File: src/content/learn/referencing-values-with-refs.md [hunk 7]
```diff
@@ -93,12 +92,12 @@ export default function Stopwatch() {
   const [now, setNow] = useState(null);
 
   function handleStart() {
-    // Start counting.
+    // Mulai menghitung
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:95]:
```suggestion
    // Mulai menghitung.
```

File: src/content/learn/referencing-values-with-refs.md [hunk 8]
```diff
@@ -93,12 +92,12 @@ export default function Stopwatch() {
...
     setStartTime(Date.now());
     setNow(Date.now());
 
     setInterval(() => {
-      // Update the current time every 10ms.
+      // Memperbarui waktu saat ini setiap 10ms.
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:100]:
```suggestion
      // Memperbarui waktu saat ini setiap 10 milidetik.
```

File: src/content/learn/referencing-values-with-refs.md [hunk 9]
```diff
@@ -121,7 +120,7 @@ export default function Stopwatch() {
 
 </Sandpack>
 
-When the "Stop" button is pressed, you need to cancel the existing interval so that it stops updating the `now` state variable. You can do this by calling [`clearInterval`](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval), but you need to give it the interval ID that was previously returned by the `setInterval` call when the user pressed Start. You need to keep the interval ID somewhere. **Since the interval ID is not used for rendering, you can keep it in a ref:**
+Ketika tombol "Berhenti" ditekan, Anda perlu membatalkan *interval* yang ada sehingga berhenti memperbarui nilai dari variabel *state* `now`. Anda dapat melakukannya dengan memanggil [`clearInterval`](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval), tetapi Anda harus memberikan *interval ID* yang sebelumnya dikembalikan oleh pemanggilan `setInterval` saat pengguna menekan "Mulai". Anda perlu menyimpan *interval ID* tersebut di suatu tempat. **Karena *interval ID* tidak digunakan untuk *rendering*, Anda dapat menyimpannya dalam *ref***:
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:123]:
```suggestion
Ketika tombol "Berhenti" ditekan, Anda perlu membatalkan *interval* yang ada sehingga komponen tersebut berhenti memperbarui nilai dari *state* `now`. Anda dapat melakukannya dengan memanggil [`clearInterval`](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval), tetapi Anda harus memberikan *interval ID* yang sebelumnya dikembalikan oleh pemanggilan `setInterval` saat pengguna menekan "Mulai". Anda perlu menyimpan *interval ID* tersebut di suatu tempat. **Karena *interval ID* tidak digunakan untuk *rendering*, Anda dapat menyimpannya dalam *ref***:
```

`` Variabel *state* `now` `` sounds redundant, that's why I suggested to shorten it to `` *state* `now` ``. Let me know what you think.

File: src/content/learn/referencing-values-with-refs.md [hunk 10]
```diff
@@ -168,20 +167,20 @@ export default function Stopwatch() {
...
+## Perbedaan antara *refs* dan *state* {/*differences-between-refs-and-state*/}
 
-Perhaps you're thinking refs seem less "strict" than state—you can mutate them instead of always having to use a state setting function, for instance. But in most cases, you'll want to use state. Refs are an "escape hatch" you won't need often. Here's how state and refs compare:
+Mungkin Anda berpikir bahwa *ref* terlihat kurang "ketat" dibandingkan dengan *state* - Anda dapat memutasi *ref* daripada selalu harus menggunakan fungsi pengaturan *state*, misalnya. Tetapi dalam kebanyakan kasus, Anda akan ingin menggunakan *state*. *Ref* adalah "jalan keluar" yang tidak sering Anda butuhkan. Berikut adalah perbandingan antara *state* dan *ref*:
 
-| refs                                                                                  | state                                                                                                                     |
+| *refs*                                                                                  | *state*                                                                                                                     |
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:176]:
```suggestion
| *ref*                                                                                  | *state*                                                                                                                     |
```

File: src/content/learn/referencing-values-with-refs.md [hunk 11]
```diff
@@ -168,20 +167,20 @@ export default function Stopwatch() {
...
-| `useRef(initialValue)` returns `{ current: initialValue }`                            | `useState(initialValue)` returns the current value of a state variable and a state setter function ( `[value, setValue]`) |
-| Doesn't trigger re-render when you change it.                                         | Triggers re-render when you change it.                                                                                    |
-| Mutable—you can modify and update `current`'s value outside of the rendering process. | "Immutable"—you must use the state setting function to modify state variables to queue a re-render.                       |
-| You shouldn't read (or write) the `current` value during rendering. | You can read state at any time. However, each render has its own [snapshot](/learn/state-as-a-snapshot) of state which does not change.
+| `useRef(initialValue)` mengembalikan `{ current: initialValue }`                            | `useState(initialValue)` mengembalikan nilai saat ini dari sebuah variabel state dan sebuah fungsi pengatur state. ( `[value, setValue]`) |
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:178]:
```suggestion
| `useRef(initialValue)` mengembalikan `{ current: initialValue }`                            | `useState(initialValue)` mengembalikan nilai saat ini dari sebuah *state* dan sebuah fungsi pengatur *state*. ( `[value, setValue]`) |
```

File: src/content/learn/referencing-values-with-refs.md [hunk 12]
```diff
@@ -168,20 +167,20 @@ export default function Stopwatch() {
...
+Ketika sebuah informasi digunakan untuk *rendering*, simpanlah di dalam *state*. Ketika sebuah informasi hanya dibutuhkan oleh *event handler* dan perubahan informasi tersebut tidak memerlukan *render* ulang, menggunakan *ref* mungkin akan lebih efisien.
 
-## Differences between refs and state {/*differences-between-refs-and-state*/}
+## Perbedaan antara *refs* dan *state* {/*differences-between-refs-and-state*/}
 
-Perhaps you're thinking refs seem less "strict" than state—you can mutate them instead of always having to use a state setting function, for instance. But in most cases, you'll want to use state. Refs are an "escape hatch" you won't need often. Here's how state and refs compare:
+Mungkin Anda berpikir bahwa *ref* terlihat kurang "ketat" dibandingkan dengan *state* - Anda dapat memutasi *ref* daripada selalu harus menggunakan fungsi pengaturan *state*, misalnya. Tetapi dalam kebanyakan kasus, Anda akan ingin menggunakan *state*. *Ref* adalah "jalan keluar" yang tidak sering Anda butuhkan. Berikut adalah perbandingan antara *state* dan *ref*:
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:174]:
```suggestion
Mungkin Anda berpikir bahwa *ref* terlihat kurang "ketat" dibandingkan dengan *state*-Anda dapat memutasi *ref* daripada selalu harus menggunakan fungsi pengaturan *state*, misalnya. Tetapi dalam kebanyakan kasus, Anda akan ingin menggunakan *state*. *Ref* adalah "jalan keluar" yang tidak sering Anda butuhkan. Berikut adalah perbandingan antara *state* dan *ref*:
```

File: src/content/learn/referencing-values-with-refs.md [hunk 13]
```diff
@@ -168,20 +167,20 @@ export default function Stopwatch() {
...
-| Mutable—you can modify and update `current`'s value outside of the rendering process. | "Immutable"—you must use the state setting function to modify state variables to queue a re-render.                       |
-| You shouldn't read (or write) the `current` value during rendering. | You can read state at any time. However, each render has its own [snapshot](/learn/state-as-a-snapshot) of state which does not change.
+| `useRef(initialValue)` mengembalikan `{ current: initialValue }`                            | `useState(initialValue)` mengembalikan nilai saat ini dari sebuah variabel state dan sebuah fungsi pengatur state. ( `[value, setValue]`) |
+| Tidak memicu *render* ulang ketika Anda mengubahnya.                                         | Memicu *render* ulang ketika Anda mengubahnya.                                                                                    |
+| *Mutable*—Anda dapat memodifikasi dan memperbarui nilai `current` di luar proses rendering. | *Immutable*—Anda harus menggunakan fungsi pengatur state untuk memodifikasi variabel state agar terjadwal ulang (*render* ulang).                       |
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:180]:
```suggestion
| *Mutable*—Anda dapat memodifikasi dan memperbarui nilai `current` di luar proses rendering. | *Immutable*—Anda harus menggunakan fungsi pengatur *state* untuk memodifikasi *state* untuk menjadwalkan *render* ulang.                       |
```

File: src/content/learn/referencing-values-with-refs.md [hunk 14]
```diff
@@ -168,20 +167,20 @@ export default function Stopwatch() {
...
+| Anda sebaiknya tidak membaca (atau menulis) nilai `current` selama proses *rendering*. | Anda dapat membaca state kapan saja. Namun, setiap *render* state memiliki [snapshot](/learn/state-as-a-snapshot) sendiri yang tidak berubah.
```
Comment (arfianadam) [inline src/content/learn/referencing-values-with-refs.md:181]:
```suggestion
| Anda sebaiknya tidak membaca (atau menulis) nilai `current` selama proses *rendering*. | Anda dapat membaca *state* kapan saja. Namun, setiap *render* terjadi *state* memiliki [*snapshot*](/learn/state-as-a-snapshot) sendiri yang tidak berubah.
```

**Review by arfianadam — COMMENTED** [review]
2nd review.

File: src/content/learn/referencing-values-with-refs.md (hunk 9 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:123]: Sure, I think it makes more sense

File: src/content/learn/referencing-values-with-refs.md (hunk 10 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:176]: is all word refs change to ref?

File: src/content/learn/referencing-values-with-refs.md (hunk 13 above)
Comment (irsyaadbp) [reply src/content/learn/referencing-values-with-refs.md:180]: I think `untuk` called 2 times sounds weird, how about
```suggestion
| *Mutable*—Anda dapat memodifikasi dan memperbarui nilai `current` di luar proses rendering. | *Immutable*—Anda harus menggunakan fungsi pengatur *state* untuk memodifikasi *state* dan menjadwalkan *render* ulang.                       |
```

Or

```suggestion
| *Mutable*—Anda dapat memodifikasi dan memperbarui nilai `current` di luar proses rendering. | *Immutable*—Anda harus menggunakan fungsi pengatur *state* untuk memodifikasi *state* dan memicu *render* ulang.                       |
```

**Comment by mhaidarhanif** [comment]
@irsyaadbp Please resolve all review comments from @arfianadam to reconfirm.

File: src/content/learn/referencing-values-with-refs.md (hunk 13 above)
Comment (arfianadam) [reply src/content/learn/referencing-values-with-refs.md]: 2nd looks better 👍

**Comment by irsyaadbp** [comment]
> @irsyaadbp Please resolve all review comments from @r17x to reconfirm all the fixed changes (by clicking all the Resolve conversation buttons). Except for one or some of them haven't been resolved.

image
@mhaidarhanif where is the button resolve conversation? but i already submit rerequest review

**Comment by arfianadam** [comment]
@irsyaadbp should be right here
image

**Review by r17x — APPROVED** [review]

**Pull request merged**